wrapanapi/systems/openstack.py

```diff
@@ -583,15 +588,23 @@ class OpenstackSystem(System, VmMixin, TemplateMixin):
     can_suspend = True
     can_pause = True
 
-    def __init__(self, **kwargs):
-        self.keystone_version = kwargs.get('keystone_version', 2)
+    def __init__(self, tenant, username, password, auth_url, **kwargs):
+        self.keystone_version = kwargs.get("keystone_version")
+        if not self.keystone_version:
+            parsed_keystone_version = search(r'v[2-3]$', kwargs['auth_url'])
```

Comment:
are you sure this works ? afaik, you need to at least add groups here. I mean ()

Reply:
@kedark3 ^^

Reply:
Yep I need to add group(), missed it. Was testing it locally and caught that too. Updating briefly.